heal/vlmd/extract/utils.py

```diff
@@ -0,0 +1,293 @@
+"""Extract utilities/helper functions"""
+
+from collections.abc import MutableMapping
+import re
+
+import pandas as pd
+from pandas.api.types import is_object_dtype
+
+
+def _get_propnames_to_rearrange(propnames, schema):
+    """
+    get data dictionary props to search for refactoring or embedding
+
+    Supports patternProperties
+    """
+
+    annotation_names = ["title", "description", "name", "additionalDescription"]
+    root_names = flatten_properties(schema["properties"]).keys()
+    field_names = flatten_properties(
+        schema["properties"]["fields"]["items"]["properties"]
+    ).keys()
+    props_to_rearrange = (
+        set(root_names).intersection(field_names).difference(annotation_names)
+    )
+
+    propnames_exp = "|".join("^" + name + "$" for name in props_to_rearrange)
+    names_to_rearrange = [name for name in propnames if re.match(propnames_exp, name)]
```

Comment:
Do we have to do `re.match()` here? I think what the `propnames_exp` and `re.match()` do here is the same as `if name in props_to_rearrange`, or maybe I'm missing something

Reply:
Yes, we still need the `re.match()`. The `propnames_exp` in the line above has regular expressions, for example `'^standardsMappings\\[\\d+\\].item.source$'`. Several functions process the regular expressions in the props names.